src/reducers/existingThoughtMove.ts

```diff
@@ -292,29 +351,18 @@ const existingThoughtMove = (state: State, { oldPath, newPath, offset }: {
     delete contextViewsNew[contextEncodedOld] // eslint-disable-line fp/no-delete
   }
 
-  /** Updates the ranks within the given path to match those in descendantUpdatesResult. */
-  const updateMergedThoughtsRank = (path: Path) => path.map(
-    (child: Child) => {
-      const updatedThought = descendantUpdatesResult[hashThought(child.value)]
-      // child.id is undefined sometimes. Unable to reproduce.
-      return {
-        ...child,
-        rank: updatedThought ? updatedThought.rank : child.rank,
-        id: child.id ?? '',
-      } as Child
-    }
-  )
-
-  // if duplicate subthoughts are merged then update rank of thoughts of cursor descendants
-  const cursorDescendantPath = (isPathInCursor && isDuplicateMerge ? updateMergedThoughtsRank : ID)(state.cursor || []).slice(oldPath.length)
+  /** Returns new path for the given old context.
+   * Note: This uses childUpdates so it works only for updated descendants. For main ancestor oldPath that has been moved (ancestor) use updatedNewPath instead.
+   */
+  const getNewPathFromOldContext = (path: Path) => childUpdates[hashContext(pathToContext(path))].pathNew
```

Comment:
I think this should take a `SimplePath` rather than a `Path`. `Contexts` are contiguous parent-child sequences and are not allowed to cross context views like `Paths` can. Unfortunately we lose our branded type safety when we call `pathToContext` (which should probably only take `SimplePath` also). You can call `simplifyPath` on `state.cursor` before passing it in.